### ISSUE TITLE
[Bug]: ERROR: Package 'networkx' requires a different Python: 3.8.10 not in '>=3.9'

### ISSUE THREAD

**Issue by ll7** [una-auxme/paf23#194]
### Current Behavior

```bash
$ b5 install
...
=> ERROR [agent 26/37] RUN source ~/.bashrc && pip install -r /workspace/requirements.txt                                                                                                                                                                                     79.7s
------                                                                                                                                                                                                                                                                               
 > [agent 26/37] RUN source ~/.bashrc && pip install -r /workspace/requirements.txt:                                                                                                                                                                                                 
1.197 Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cu118                                                                                                                                                                                            
1.986 Collecting dvc==3.30.3                                                                                                                                                                                                                                                         
2.035   Downloading dvc-3.30.3-py3-none-any.whl (438 kB)                                                                                                                                                                                                                             
2.668 Collecting dvc-gdrive==2.20.0                                                                                                                                                                                                                                                  
2.718   Downloading dvc_gdrive-2.20.0-py3-none-any.whl (11 kB)
3.485 Collecting torch==2.1.1
3.499   Downloading https://download.pytorch.org/whl/cu118/torch-2.1.1%2Bcu118-cp38-cp38-linux_x86_64.whl (2325.9 MB)
63.23 Collecting torchvision==0.16.1
63.24   Downloading https://download.pytorch.org/whl/cu118/torchvision-0.16.1%2Bcu118-cp38-cp38-linux_x86_64.whl (6.1 MB)
64.25 Collecting torchaudio==2.1.1
64.27   Downloading https://download.pytorch.org/whl/cu118/torchaudio-2.1.1%2Bcu118-cp38-cp38-linux_x86_64.whl (3.2 MB)
65.01 Collecting pytorch-lightning==2.1.2
65.02   Downloading pytorch_lightning-2.1.2-py3-none-any.whl (776 kB)
65.80 Collecting opencv-python==4.8.1.78
65.81   Downloading opencv_python-4.8.1.78-cp37-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (61.7 MB)
68.03 Collecting dvclive==3.3.1
68.05   Downloading dvclive-3.3.1-py3-none-any.whl (34 kB)
68.96 Collecting ruamel.yaml==0.18.5
68.97   Downloading ruamel.yaml-0.18.5-py3-none-any.whl (116 kB)
69.76 Collecting scipy==1.10.1
69.78   Downloading scipy-1.10.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (34.5 MB)
71.40 Collecting xmltodict==0.13.0
71.42   Downloading xmltodict-0.13.0-py2.py3-none-any.whl (10.0 kB)
71.90 Collecting py-trees==2.2.3
71.92   Downloading py_trees-2.2.3-py3-none-any.whl (113 kB)
72.84 Collecting numpy==1.23.5
72.86   Downloading numpy-1.23.5-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.1 MB)
74.11 Collecting ultralytics==8.0.220
74.13   Downloading ultralytics-8.0.220-py3-none-any.whl (645 kB)
74.93 Collecting scikit-learn>=0.18
74.95   Downloading scikit_learn-1.3.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.1 MB)
76.25 Collecting pandas==2.0.0
76.26   Downloading pandas-2.0.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.3 MB)
76.93 Requirement already satisfied: tabulate>=0.8.7 in /home/carla/.local/lib/python3.8/site-packages (from dvc==3.30.3->-r /workspace/requirements.txt (line 1)) (0.9.0)
77.40 Collecting networkx>=2.5
77.41   Downloading https://download.pytorch.org/whl/networkx-3.2.1-py3-none-any.whl (1.6 MB)
79.09 ERROR: Package 'networkx' requires a different Python: 3.8.10 not in '>=3.9'
------
failed to solve: process "/bin/bash -c source ~/.bashrc && pip install -r /workspace/requirements.txt" did not complete successfully: exit code: 1

Error: '"$@"' exited with status 17
b5 exiting with status 1

Task failed, see error above

Error: '( export COMPOSE_PROJECT_NAME=paf23; export DOCKER_HOST_SYSTEM=linux; export DOCKER_HOST_USERNAME=luttkule; export DOCKER_HOST_UNIX_UID=1000; export DOCKER_HOST_UNIX_GID=999; "$@" )' exited with status 1
b5 exiting with status 1

Task failed, see error above

Error: '( cd /home/luttkule/git/paf23/build && docker:run docker compose -f docker-compose.yml -f docker-compose.nvidia.yml "$@" )' exited with status 1
b5 exiting with status 1

Task failed, see error above
Task execution failed, see above
```

### Expected Behavior

_No response_

### How to reproduce the issue

b5 install

**Comment by ll7** [comment]
https://github.com/una-auxme/paf23/blob/main/code/requirements.txt does not include `networkx` as a requirement.

```bash
77.40 Collecting networkx>=2.5
77.41   Downloading https://download.pytorch.org/whl/networkx-3.2.1-py3-none-any.whl (1.6 MB)
79.09 ERROR: Package 'networkx' requires a different Python: 3.8.10 not in '>=3.9'
------
failed to solve: process "/bin/bash -c source ~/.bashrc && pip install -r /workspace/requirements.txt" did not complete successfully: exit code: 1
```
Does not indicate where the requirement comes from.

**Comment by MaxJa4** [comment]
Fixed by #193
